[Wee Hung] iP

WeeeHung opened the pull request:
### **iPBot**

> "Believe you can and you're halfway there." - Theodore Roosevelt

## Introduction 
As the name suggests, this is a bot made for _Individual Project (IP)_. This is a project that gets improved and updated weekly during the semester.

## Features
Some functionalities are as follows:
- listing out todo list
- setting deadline of tasks
- jotting down events

## Contributions
The project is inspired from [https://nus-cs2103-ay2324s1.github.io/website/admin/](url)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

WeeeHung committed:
Level-0: Rename, Greet, Exit

WeeeHung committed:
Level-1: Echo

WeeeHung committed:
Level-2: Add, List

WeeeHung committed:
Level-3: Mark as Done

WeeeHung committed:
Level-4: ToDo, Event, Deadline

WeeeHung committed:
Level 6: Delete

WeeeHung committed:
Level 7. Save

WeeeHung committed:
Merge pull request #1 from WeeeHung/branch-level-7

Level 7. Save

WeeeHung committed:
Level-8: Dates and Times

WeeeHung committed:
test commit

WeeeHung committed:
add UI Package

WeeeHung committed:
Add package for iPBot

WeeeHung committed:
Merge remote-tracking branch 'origin/add-gradle-support'

WeeeHung committed:
A-JUnit

WeeeHung committed:
Add Javadocs

WeeeHung committed:
Edit to fit Java Coding Standard

WeeeHung committed:
Edit to fit Java Coding Standar

ZD292 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,117 @@
+     * @param args The command-line arguments passed to the application.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
```
Do you want to print the logo using ChatUI class instead?

WeeeHung replied:
oops Thank you. I missed this out and forgot to remove logo. Current UI doesn't include logo

ZD292 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,117 @@
+        storage.loadListFromFile();
+        ui.printIntro();
+
+        while (true) {
```
Clean and easy to read 😄

WeeeHung replied:
Thank you

ZD292 commented on src/main/java/DukePackage/ChatUI.java:
```diff
@@ -0,0 +1,42 @@
+package DukePackage;
```
Looks good 😸

WeeeHung replied:
Thank you

ZD292 commented on src/main/java/DukePackage/Parser.java:
```diff
@@ -0,0 +1,34 @@
+    public String getInput() {
+        Scanner scanner = this.scanner;
+
+        // Prompt the user for input
```
Perhaps inline comments can be added under JavaDocs description instead

WeeeHung replied:
Valid! noted with thanks.

ZD292 commented on src/main/java/DukePackage/Storage.java:
```diff
@@ -0,0 +1,224 @@
+    /**
+     * Writes the task list to a file.
+     */
+    public void writeTasksToFile() {
```
I like how you implemented this

WeeeHung replied:
thank you.

ZD292 commented on src/main/java/DukePackage/Task.java:
```diff
@@ -0,0 +1,144 @@
+package DukePackage;
```
Looks good 👍

WeeeHung replied:
thank you.

ZD292 reviewed (commented):
I like how you implemented the methods 🥇

Sebtey commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,117 @@
+            String[] parts = input.split(" ");
+            ui.printDivider();
+
+            switch (parts[0]) {
```
Hey, I may be wrong, but according to checkstyle, the indentations for the case lines are supposed to be the same as that of switch, while the codes for each case is supposed to be one indentation more than the case statements.

Sebtey commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,117 @@
+                        storage.changeTaskMarking(id, true);
+                        storage.printTaskMarking(id);
+                    } catch (Exception e) {
+                        System.out.print(e.getMessage());
```
Hmm, perhaps you can encapsulate the printing of exceptions thrown under ChatUI class as well?

Sebtey commented on src/main/java/DukePackage/Storage.java:
```diff
@@ -0,0 +1,224 @@
+    /**
+     * Writes the task list to a file.
+     */
+    public void writeTasksToFile() {
```
This method is currently pretty long, would it be possible to compress its implementations into fewer lines by further abstracting out the different parts of it such as the for-loop?

Sebtey commented on src/main/java/DukePackage/Storage.java:
```diff
@@ -0,0 +1,224 @@
+     */
+    public void printTaskList() {
+        for (int i = 0; i < this.taskList.size(); i++) {
+            int index = i + 1;
```
Think it would be great if you could add a comment here explaining the rationale for + 1! 😺

Sebtey reviewed (commented):
Overall, I find it relatively manageable to follow the codes and the JavaDocs seems to be succinct! 👍 

Some suggestions that I have that you could perhaps consider would be:

1.  To further encapsulate the System.out.println statements under ChatUi class.
2. Create more Exception classes extending DukeException, for specific errors

WeeeHung committed:
Level 10: GUI

WeeeHung committed:
done building jar files

WeeeHung committed:
ChatGUI process method: carry on without storage

invalid storage will lead to error in process method

Java Assertion used to terminate program if storage is invalid

Prevent program from crashing in unexpected ways
if storage is invalid

WeeeHung committed:
Add Code Quality for repo

Some functions were unnecessarily long and complicated,
so I refactored them to be shorter and more readable.

Shorten functions and removed arrow-headed functions,
incorporated SLAP principle and renamed variables and packages
to be more readable.

WeeeHung committed:
Merge pull request #3 from WeeeHung/A-CodeQuality

Add Code Quality for repo

WeeeHung committed:
Extension to boost functionality of chat bot.

Add help command to display all commands available.

WeeeHung committed:
Merge pull request #4 from WeeeHung/branch-BCD-Extension

Extension to boost functionality of chat bot.

WeeeHung committed:
Add ui.png

WeeeHung committed:
Merge branch 'master' of https://github.com/WeeeHung/ip

WeeeHung committed:
Add UserGuide to ReadMe in GFMD

WeeeHung committed:
Merge pull request #5 from WeeeHung/branch-A-UserGuide

Add UserGuide to ReadMe in GFMD

WeeeHung committed:
Add UserGuide to ReadMe in GFMD

WeeeHung committed:
Merge pull request #6 from WeeeHung/branch-A-UserGuide

Add UserGuide to ReadMe in GFMD

WeeeHung committed:
Change Ui.png file name

WeeeHung committed:
Merge pull request #7 from WeeeHung/branch-A-UserGuide

Change Ui.png file name

WeeeHung committed:
Amend for jar file release

WeeeHung committed:
Update user guide

WeeeHung committed:
Release iPBot v0.1

WeeeHung committed:
Release iPBot v0.2

WeeeHung committed:
Touch up on tags

WeeeHung committed:
ammend changes to clean dead code

WeeeHung committed:
Merge pull request #9 from WeeeHung/branch-FixIP

ammend changes to clean dead code
